### PR TITLE
set [tool.setuptools] package-dir to force setuptools editables to append to sys.path

### LINKED ISSUE #7782: Missing `__version__` and other attributes
I have an editable version of `distributed` installed locally with `pip install -e .` from the root of the repo. When I drop into `ipython` I see

```python
In [1]: import distributed
distr
In [2]: distributed.__file__
Out[2]: '/Users/james/projects/dask/distributed/distributed/__init__.py'

In [3]: distributed.__version__
Out[3]: '2023.4.0+1.g5c0a412c'
```

This looks like I expect it to. When I then navigate to another directory, for example, over to the root of `dask/dask` on my laptop, I no longer see the same `distributed.__version__` and `distributed.__file__`:

```python
In [1]: import distributed

In [2]: distributed.__file__

In [3]: distributed.__version__
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Input In [3], in <cell line: 1>()
----> 1 distributed.__version__

AttributeError: module 'distributed' has no attribute '__version__'
```

I've not run `git bisect`, but my guess is https://github.com/dask/distributed/pull/7629 is related

cc @graingert for visibility

### PR DESCRIPTION
**Pull request by graingert**
Closes #7782

- [x] Tests added / passed
- [x] Passes `pre-commit run --all-files`

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -62,6 +62,7 @@ dask-worker = "distributed.cli.dask_worker:main"
 [tool.setuptools]
 include-package-data = true
 zip-safe = false
+package-dir = {"" = "."}
```
Comment (graingert) [inline pyproject.toml:65]: this causes setuptools to go down the same codepath as `is_compat_mode`:

https://github.com/pypa/setuptools/blob/e94cb204693cee145ea531905dd50b02a26c9f0a/setuptools/command/editable_wheel.py#L376-L378

```
        if set(self.package_dir) == {""} and has_simple_layout or is_compat_mode:
            # src-layout(ish) is relatively safe for a simple pth file
            src_dir = self.package_dir.get("", ".")
```

**Comment by fjetter** [comment]
I'm OK with this fix for now. @graingert offline we discussed briefly why the repo is not detected as simple. Can you summarize this here briefly for prosperity in case this hack fails us?

**Comment by graingert** [comment]
@fjetter ah it's no good, this breaks `include-package-data` and so `distributed.yaml` does not get included in the built wheel
